inception-recommendation/src/main/java/de/tudarmstadt/ukp/inception/recommendation/project/RecommenderEditorPanel.java

```diff
@@ -347,12 +347,15 @@ private void autoUpdateName(AjaxRequestTarget aTarget, TextField<String> aField,
     }
     
     /**
-     * Check if the selected recommender still accepts the configured layer and feature.
-     * If not show an error message.
+     * Check if the selected recommender still accepts the configured layer and feature. If not show
+     * an error message.
      */
     private void checkRecommenderLayerMatch(IModel<Pair<String, String>> aToolModel)
     {
-        if (recommenderModel.getObject() != null && aToolModel.getObject() != null) {
+        if (recommenderModel.getObject() == null || aToolModel.getObject() == null) {
```

Comment:
Since you return in the "if" block, the "else" block is not required. Cf. http://blog.timoxley.com/post/47041269194/avoid-else-return-early

Reply:
Thx